app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchRepositoryImpl.java

```diff
@@ -589,6 +598,22 @@ else if (d2.trackedEntityModule().trackedEntityAttributes().byUid().eq(tetAttrib
         return teis;
     }
 
+    private List<TrackedEntityInstance> filterDeletedEnrollment(List<TrackedEntityInstance> teis, @Nullable Program program) {
+        Iterator<TrackedEntityInstance> iterator = teis.iterator();
+        while (iterator.hasNext()) {
+            TrackedEntityInstance tei = iterator.next();
+            if (d2.trackedEntityModule().trackedEntityInstances().byUid().eq(tei.uid()).one().blockingExists()) {
+                if (program != null) {
+                    Enrollment enrollment = d2.enrollmentModule().enrollments().byTrackedEntityInstance().eq(tei.uid()).byProgram().eq(program.uid()).one().blockingGet();
+                    if (enrollment.program().equals(program.uid()) && enrollment.deleted()) {
+                        iterator.remove();
+                    }
+                }
+            }
+        }
+        return teis;
+    }
```

Comment:
to me this should go outside repository, it's not its responsability

Reply:
thoughts? @ferdyrod

Reply:
Yes, it should. this whole repository needs a lot of rework.

Reply:
:+1:  great ! do you think this small piece can be extracted (only filterDeletedEnrollment) or has other considerations??

Reply:
I'm going to extract it to an extension function